File: ShowDown_Java/Game.java
package ShowDown_Java;

import java.util.ArrayList;
import java.util.List;

public class Game {
    private int TOTAL_PLAYER_NUM = 4;
    private int HUMAN_PLAYER_NUM = 1;

    public Game(){
    
    }
    
    public void initial() {
        List<Player> players = generatePlayers();
        for (Player player : players) {
            player.nameHimself();
        }
        System.out.println("Players: " + players);
    }

    private List<Player> generatePlayers() {
        List<Player> players = new ArrayList<>();
        
        for (int i = 0; i < HUMAN_PLAYER_NUM; i++) {
            HumanPlayer humanPlayer = new HumanPlayer();
            players.add(humanPlayer);
        }

        int AIPlayerNum = TOTAL_PLAYER_NUM - HUMAN_PLAYER_NUM;
        for (int i = 0; i < AIPlayerNum; i++) {
            AIPlayer AIPlayer = new AIPlayer();
            players.add(AIPlayer);
        }

        return players;
    }
}

File: ShowDown_test/Game.js
const { HumanPlayer, AIPlayer } = require('./Player');

class Game{
    constructor(){
        this.players = [];
    }

    initial(){
        const playerTotalNum = 4, numOfHuman = 1;
        const numOfAI = playerTotalNum - numOfHuman;
        this.addPlayers(numOfHuman, numOfAI);
        this.start();       
    }

    addPlayers(numOfHuman, numOfAI){
        console.log("line#16");
        // for(let i = 0 ; i < numOfHuman; i++){
        //     console.log("line#18");
        //     //const humanPlayer = new HumanPlayer();
        //     humanPlayer.nameHimself();
        //     this.players.push(humanPlayer);
        //     console.log("human player: " + humanPlayer)
        // }
        for (let i = 0; i < numOfAI; i++) {
            const aiPlayer = new AIPlayer();
            aiPlayer.nameHimself();
            this.players.push(aiPlayer);
        }
    }

    start(){
        console.log("Game starting with players:");
        this.players.forEach((player, index) => {
            const playerName = player instanceof HumanPlayer ? `Human Player ${index + 1}` : `AI Player ${index + 1}`;
            console.log(playerName);
        });
    }

}

module.exports = Game;